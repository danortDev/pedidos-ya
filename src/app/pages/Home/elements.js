import { styled } from 'styletron-react';

export const Title = styled('h1', {
  fontWeigth: '500',
  fontSize: '1.5em',
  textAlign: 'center',
  margin: '20px'
});

export const ButtonWrapper = styled('div', {
  display: 'flex',
  justifyContent: 'center',
  margin: '20px'
});
