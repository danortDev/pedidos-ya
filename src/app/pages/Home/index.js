import React, { Component } from 'react';
import objectToQueryString from 'app/utils/objectToQueryString';
import Map from 'app/components/ui/Map';
import Button from 'app/components/ui/Button';
import { Title, ButtonWrapper } from './elements';

const DEFAULT_CENTER = {
  lat: -34.9209098815918,
  lng: -56.150474548339844
};

class Home extends Component {
  state = { lat: null, lng: null };

  componentDidMount() {
    if (navigator.geolocation) {
      navigator.geolocation.getCurrentPosition((position) => {
        const { coords: { latitude: lat, longitude: lng } } = position;
        this.setCenter({ lat, lng });
      });
    } else {
      this.setCenter(DEFAULT_CENTER);
    }
  }

  setCenter = ({ lat, lng }) => {
    this.setState({ lat, lng });
  }

  search = () => {
    const { history } = this.props;
    const { lat, lng } = this.state;
    const query = objectToQueryString({ lat, lng });
    history.push(`/results${query}`);
  }

  render() {
    const { lat, lng } = this.state;
    return (
      <div>
        <Title>
          Search for restaurants near to you!
        </Title>
        <Map
          handleClick={this.setCenter}
          center={{ lat, lng }}
          zoom={14}
        />
        <ButtonWrapper>
          <Button onClick={this.search}>
            Search
          </Button>
        </ButtonWrapper>
      </div>
    );
  }
}

export default Home;
