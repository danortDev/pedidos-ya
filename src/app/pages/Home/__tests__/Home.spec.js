import React from 'react';
import Home from '../';
import { shallow } from 'enzyme';


describe('<Home/>', () => {
  it('should render', () => {
    const wrapper = shallow(<Home />);
    expect(wrapper).toMatchSnapshot();
  });
});
