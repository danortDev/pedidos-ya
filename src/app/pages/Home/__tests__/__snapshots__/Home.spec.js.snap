// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home/> should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Styled(h1)>
          Search for restaurants near to you!
        </Styled(h1)>,
        <Map
          center={
            Object {
              "lat": 50,
              "lng": 33,
            }
          }
          handleClick={[Function]}
          zoom={14}
        />,
        <Styled(div)>
          <Button
            onClick={[Function]}
          >
            Search
          </Button>
        </Styled(div)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Search for restaurants near to you!",
        },
        "ref": null,
        "rendered": "Search for restaurants near to you!",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "center": Object {
            "lat": 50,
            "lng": 33,
          },
          "handleClick": [Function],
          "zoom": 14,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Button
            onClick={[Function]}
          >
            Search
          </Button>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Search",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Search",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Styled(h1)>
            Search for restaurants near to you!
          </Styled(h1)>,
          <Map
            center={
              Object {
                "lat": 50,
                "lng": 33,
              }
            }
            handleClick={[Function]}
            zoom={14}
          />,
          <Styled(div)>
            <Button
              onClick={[Function]}
            >
              Search
            </Button>
          </Styled(div)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Search for restaurants near to you!",
          },
          "ref": null,
          "rendered": "Search for restaurants near to you!",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "center": Object {
              "lat": 50,
              "lng": 33,
            },
            "handleClick": [Function],
            "zoom": 14,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Button
              onClick={[Function]}
            >
              Search
            </Button>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Search",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Search",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
